ipnsfs/dir.go

```diff
@@ -0,0 +1,347 @@
+package ipnsfs
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"sync"
+
+	dag "github.com/jbenet/go-ipfs/merkledag"
+	ft "github.com/jbenet/go-ipfs/unixfs"
+	ufspb "github.com/jbenet/go-ipfs/unixfs/pb"
+)
+
+var ErrNotYetImplemented = errors.New("not yet implemented")
+var ErrInvalidChild = errors.New("invalid child node")
+
+type Directory struct {
+	fs        *Filesystem
+	parent    childCloser
+	childDirs map[string]*Directory
+	files     map[string]*file
+
+	node *dag.Node
+	name string
+	lock sync.Mutex
```

Comment:
i like the style of having the `mu` above what it is protecting.

Also we have `lock` and `refLock` here. maybe a more descriptove for `lock`? like `nodeMu` and `refMu` ?